interface/billing/sl_receipts_report.php

```diff
@@ -413,38 +431,50 @@ function sel_diagnosis() {
       // If a doctor was specified.
       if ($form_doctor) $query .= " AND u.id = '$form_doctor'";
       ***************************************************************/
+      $sqlBindArray = array();
       $query = "SELECT a.pid, a.encounter, a.post_time, a.code, a.modifier, a.pay_amount, " .
         "fe.date, fe.id AS trans_id, fe.provider_id AS docid, fe.invoice_refno, s.deposit_date, s.payer_id, " .
-        "b.provider_id " .
+        "b.provider_id, concat(lname, ' ', fname) as 'fulname', lname as 'last', fname as 'first' " .
         "FROM ar_activity AS a " .
         "JOIN form_encounter AS fe ON fe.pid = a.pid AND fe.encounter = a.encounter " .
         "LEFT OUTER JOIN ar_session AS s ON s.session_id = a.session_id " .
+        "LEFT OUTER JOIN patient_data ON patient_data.pid = a.pid " .
         "LEFT OUTER JOIN billing AS b ON b.pid = a.pid AND b.encounter = a.encounter AND " .
         "b.code = a.code AND b.modifier = a.modifier AND b.activity = 1 AND " .
         "b.code_type != 'COPAY' AND b.code_type != 'TAX' " .
         "WHERE a.pay_amount != 0 AND ( " .
-        "a.post_time >= '$form_from_date 00:00:00' AND a.post_time <= '$form_to_date 23:59:59' " .
-        "OR fe.date >= '$form_from_date 00:00:00' AND fe.date <= '$form_to_date 23:59:59' " .
-        "OR s.deposit_date >= '$form_from_date' AND s.deposit_date <= '$form_to_date' )";
+        "a.post_time >= ? AND a.post_time <= ? " .
+        "OR fe.date >= ? AND fe.date <= ? " .
+        "OR s.deposit_date >= ? AND s.deposit_date <= ? )";
+        array_push($sqlBindArray,$from_date,$to_date,$from_date,$to_date,$form_from_date,$form_to_date);
       // If a procedure code was specified.
       // Support code type if it is in the ar_activity table. Note it is not always included, so
       // also support a blank code type in ar_activity table.
-      if ($form_proc_codetype && $form_proc_code) $query .= " AND (a.code_type = '$form_proc_codetype' OR a.code_type = '') AND a.code = '$form_proc_code'";
+      #if ($form_proc_codetype && $form_proc_code) $query .= " AND (a.code_type = '$form_proc_codetype' OR a.code_type = '') AND a.code = '$form_proc_code'";
+      if ($form_proc_codetype && $form_proc_code) {
+        $query .= " AND (a.code_type = '$form_proc_codetype' OR a.code_type = '') AND a.code = '$form_proc_code'";
+        array_push($sqlBindArray,$form_proc_codetype,$form_proc_code);
```

Comment:
This will break and error out since not binded above. Make sure you test your code, which should catch these things.